src/modules/services/vault.nix

```diff
@@ -0,0 +1,121 @@
+{ pkgs, lib, config, ... }:
+
+let
+  cfg = config.services.vault;
+
+  types = lib.types;
+
+  configFile = pkgs.writeText "config.hcl" ''
+    storage "file" {
+      path = "${config.env.DEVENV_STATE}/vault-data"
+    }
+
+    listener "tcp" {
+      address     = "${cfg.address}"
+      tls_disable = "true"
+    }
+
+    disable_mlock      = ${if cfg.disableMlock then "true" else "false"}
+    disable_clustering = ${if cfg.disableClustering then "true" else "false"}
+    ui                 = ${if cfg.ui then "true" else "false"}
+  '';
+
+  configureScript = pkgs.writeShellScriptBin "configure-vault" ''
+    set -euo pipefail
+
+    # Wait for the vault server to start up
+    response=""
+    while [ -z "$response" ]; do
+      response=$(${pkgs.curl}/bin/curl -s --max-time 5 "${config.env.VAULT_API_ADDR}/v1/sys/init" | ${pkgs.jq}/bin/jq '.initialized' || true)
+      if [ -z "$response" ]; then
+        echo "Waiting for server to respond..."
+        sleep 1
+      fi
+    done
+
+    if [ -f "${config.env.DEVENV_STATE}/env_file" ]; then
+      source "${config.env.DEVENV_STATE}/env_file"
+    fi
+
+    # Initialize it if needed
+    if [ "$response" == "false" ]; then
+      echo "Server is not initialized. Running the script..."
+      response=$(${pkgs.curl}/bin/curl -s --request POST --data '{"secret_shares": 1, "secret_threshold": 1}' "${config.env.VAULT_API_ADDR}/v1/sys/init")
+
+      root_token=$(echo "$response" | ${pkgs.jq}/bin/jq -r '.root_token')
+      first_key_base64=$(echo "$response" | ${pkgs.jq}/bin/jq -r '.keys_base64[0]')
+
+      export VAULT_TOKEN="$root_token"
+      export UNSEAL_KEY="$first_key_base64"
+
+      echo "export VAULT_TOKEN=$VAULT_TOKEN" > "${config.env.DEVENV_STATE}/env_file"
+      echo "export UNSEAL_KEY=$UNSEAL_KEY" >> "${config.env.DEVENV_STATE}/env_file"
+    else
+      echo "Server is initialized. Skipping the script execution."
+    fi
+
+    echo "Vault Unseal key is $UNSEAL_KEY"
+    echo "Vault Root token is $VAULT_TOKEN"
+
+    while true
+    do
+      sleep 1
+    done
+  '';
+in
+{
+  imports = [
+    (lib.mkRenamedOptionModule [ "vault" "enable" ] [ "services" "vault" "enable" ])
```

Comment:
This isn't needed since we never had `vault.enable`

Reply:
Thanks for the feedback, I've removed that in the new commit.